### PR TITLE
new hook-in for falcor legacy polyfills

### PR DESCRIPTION
**Pull request by asyncanup**

### CODE REVIEW

File: lib/get/get.js [hunk 1]
```diff
@@ -63,7 +63,13 @@ module.exports = function get(walk, isJSONG) {
                     criticalError: new BoundJSONGraphModelError()
                 };
             }
-            currentCachePosition = getCachePosition(model, boundPath);
+
+            // using _getOptimizedPath because that's a point of extension
+            // for polyfilling legacy falcor
+            optimizedPath = model._getOptimizedBoundPath();
```
Comment (michaelbpaulson) [inline lib/get/get.js]: +1, good hook.

File: lib/Model.js [hunk 2]
```diff
@@ -141,6 +141,16 @@ Model.prototype._collectRatio = 0.75;
  */
 Model.prototype.get = require("./response/get");
 
+Model.prototype._getOptimizedBoundPath = function _getOptimizedBoundPath() {
```
Comment (michaelbpaulson) [inline lib/Model.js]: Don't we have a copy already?

Comment (asyncanup) [reply lib/Model.js]: Copy of what?

Comment (michaelbpaulson) [reply lib/Model.js]: https://github.com/Netflix/falcor/blob/master/lib/support/array-clone.js

Comment (michaelbpaulson) [reply lib/Model.js]: I would imagine this function being a touch more like 
```
const clone = require('./path/to/array-clone');

Model.prototype._getOptimizedBoundPath = function _getOptimizedBoundPath() {
    return clone(this._path);
}
```

Comment (asyncanup) [reply lib/Model.js]: ah cool

**Comment by michaelbpaulson** [comment]
lgtm

**Pull request merged**